Sources/Workspace/PinsStore.swift

```diff
@@ -157,15 +147,14 @@ public final class PinsStore {
                 container: pin.repository, requirement: pin.state.requirement())
         }
     }
+    
+    public func saveState() throws {
```

Comment:
Please add a simple Doc comment, something like this should be enough.
`/// Writes the pins to pins file.`